docs/openapi.yaml

```diff
@@ -0,0 +1,1143 @@
+      summary: Create new event for robot
+      description: |
+        ### Overview
+        Adds a new event to the robot schedule
+
+        New entries to the schedule can be added as long as they don't conflict with already scheduled events.
+      tags:
+        - Robots
+      parameters:
+        - in: path
+          name: robot_id
+          required: true
+          schema:
+            type: integer
+            format: int32
+            example: 21
+      requestBody:
+        description: Time entry update
+        required: true
+        content:
+          application/json:
+            schema:
+              $ref: "#/components/schemas/EventResponse"
+            examples:
+              scheduleEvent:
+                $ref: "#/components/examples/scheduleEvent"
+      responses:
+        "200":
+          content:
+            application/json:
+              schema:
+                $ref: "#/components/schemas/EventResponse"
+          description: Request successful
+        "403":
+          description: Forbidden - Client not authorized to add events
+          content:
+            application/json:
+              schema:
+                $ref: "#/components/schemas/ProblemDetails"
+        default:
+          description: Unexpected error
+          content:
+            application/json:
+              schema:
+                $ref: "#/components/schemas/ProblemDetails"
+
+  /robots/{robot_id}/event/{event_id}:
+    delete:
+      operationId: deleteRobotEvent
+      summary: Remove an event from the robot schedule
+      description: |
+        ### Overview
+        Deletes an event from the robot schedule.
+
+        Can only be used for events that have not started yet.
+      tags:
+        - Robots
+      parameters:
+        - in: path
+          name: robot_id
+          required: true
+          schema:
+            type: integer
+            format: int32
+            example: 21
+        - in: path
+          name: event_id
+          required: true
+          schema:
+            type: integer
+            format: int32
+            example: 102
+      responses:
+        "200":
+          content:
+            application/json:
+              schema:
+                $ref: "#/components/schemas/PostResponse"
+          description: Request successful
+        "403":
+          description: Forbidden - Client not authorized to delete events
+          content:
+            application/json:
+              schema:
+                $ref: "#/components/schemas/ProblemDetails"
+        default:
+          description: Unexpected error
+          content:
+            application/json:
+              schema:
+                $ref: "#/components/schemas/ProblemDetails"
+
+  /robots/{robot_id}/start/{mission_id}:
+    post:
+      operationId: postStartRobot
+      summary: Start a mission with robot
+      description: |
+        ### Overview
+        Start a mission with given id using robot with robot id.
```

Comment:
Do we start it immediatly and/or at a given time?

Reply:
I think it should start immediately and create event in the case we want to start it at a given time.